src/Symfony/Component/Serializer/Encoder/JsonDecode.php

```diff
@@ -78,6 +81,16 @@ public function decode($data, $format, array $context = array())
             throw new NotEncodableValueException(json_last_error_msg());
         }
 
+        if ($rootKey = $context['json_root_key']) {
```

Comment:
We try to use public constants for context keys, so they are discoverable & autocompletable. #28709 is even completing missing ones.

Reply:
Thanks for the review @ogizanagi.  All comments resolved.